c++/src/kj/string.h

```diff
@@ -75,6 +75,10 @@ class StringPtr {
   }
   inline StringPtr(const char* begin, const char* end): StringPtr(begin, end - begin) {}
   inline StringPtr(const String& value);
+  void operator=(String&& value) = delete;
+  inline void operator=(decltype(nullptr)) {
```

Comment:
required to resolve an operator ambiguity since `String` can be implicitly constructed from `nullptr`, and we actually do want to allow `StringPtr s = nullptr`